feat: Update RN troubleshooting docs for promise polyfill fix (#4525)
getsentry/sentry-docs: src/platforms/react-native/troubleshooting.mdx

## Unhandled Promise Rejections

Due to an issue with React Native's dependencies, unhandled promise rejections might not be correctly caught by Sentry. If the promise rejection handler was not correctly attached, our SDK might issue a warning:

> WARN: Unhandled promise rejections might not be caught by Sentry. Read about how to fix this on our troubleshooting docs.

Otherwise, we will let you know that the handler is attached:

> [Sentry] Unhandled promise rejections will be caught by Sentry.

### Auto Patching (Default Behavior)

By default we will patch the global `Promise` instance to ensure it matches exactly with the version that React Native uses.

#### Using With Other Polyfills

> If you use some polyfilling library that patches the global `Promise` instance, you will need to make sure you run the polyfill **after** `Sentry.init` is called.

Reviewer: ```suggestion
If you use a polyfilling library that patches the global `Promise` instance, you'll need to make sure you run the polyfill **after** `Sentry.init` is called.
```

```js
import allSettled from "promise.allsettled";

Sentry.init({
  dsn: "___PUBLIC_DSN___",
});
// Any Promise polyfilling must occur AFTER Sentry.init
// This step globally patches Promise.
allSettled.shim();

// Separate core-js example
import "core-js/stable/promise/all-settled";
```

> Your linter might throw some errors at you here, but this step is necessary.

Reviewer: ```suggestion
Your linter might throw some errors here, but this step is necessary.
```

#### Disable Auto Patching

> You can disable the global promise patching by passing `patchGlobalPromise: false` either in `Sentry.init` or the `ReactNativeErrorHandlers` integration. Note that if you disable our auto patching, to ensure that unhandled rejections are still caught you will need to [manually force a package resolution](#manually-forcing-a-package-resolution);

Reviewer 2: Worth adding this option to the Configuration page too.

> You can disable the global promise patching by passing `patchGlobalPromise: false` either in `Sentry.init` or the `ReactNativeErrorHandlers` integration. Note that if you disable our auto patching, to ensure that unhandled rejections are still caught you will need to [manually force a package resolution](#manually-forcing-a-package-resolution);

Reviewer: ```suggestion
You can disable the global promise patching by passing `patchGlobalPromise: false` in either `Sentry.init` or the `ReactNativeErrorHandlers` integration. Note that if you disable our auto patching, to ensure that unhandled rejections are still caught, you will need to [manually force a package resolution](#manually-forcing-a-package-resolution).
```

### Manually Forcing a Package Resolution

> You will not need to perform the steps below if you do not disable [auto patching](#auto-patching-default-behavior). You will need to ensure that the version of `promise` that you use matches exactly with the version that React Native uses.

Reviewer: ```suggestion
You don't need to perform the steps below if you don't disable [auto patching](#auto-patching-default-behavior). You'll need to ensure that the version of `promise` that you use matches exactly with the version that React Native uses.
```

</Note>

> 3. If the fix is successful, our SDK will no longer display the above warning and will say promise rejections will be caught.

Reviewer: ```suggestion
3. If the fix is successful, our SDK will no longer display the above warning and will indicate that promise rejections will be caught.
```
